pkg/azurefile/azurefile.go

```diff
@@ -870,6 +881,60 @@ func (d *Driver) ResizeFileShare(ctx context.Context, subsID, resourceGroup, acc
 	})
 }
 
+// CopyFileShare copies a fileshare in the same storage account
+func (d *Driver) CopyFileShare(ctx context.Context, req *csi.CreateVolumeRequest, accountKey string, shareOptions *fileclient.ShareOptions, storageEndpointSuffix string) error {
+	if shareOptions.Protocol == storage.EnabledProtocolsNFS {
+		return fmt.Errorf("protocol nfs is not supported for volume cloning")
+	}
+	var volumeID string
+	if req.GetVolumeContentSource() != nil && req.GetVolumeContentSource().GetVolume() != nil {
+		volumeID = req.GetVolumeContentSource().GetVolume().GetVolumeId()
+	}
+	resourceGroupName, accountName, srcFileShareName, _, _, subsID, err := GetFileShareInfo(volumeID)
+	dstFileShareName := shareOptions.Name
+	if err != nil {
+		return status.Error(codes.NotFound, err.Error())
+	}
+	klog.V(2).Infof("begin to copy fileshare %s to %s", srcFileShareName, dstFileShareName)
+	if srcFileShareName == "" {
+		return fmt.Errorf("srcFileShareName is empty")
+	}
+	if dstFileShareName == "" {
+		return fmt.Errorf("dstFileShareName is empty")
+	}
+
+	klog.V(2).Infof("generate sas token for account(%s)", accountName)
+	accountSasToken, genErr := generateSASToken(accountName, accountKey, storageEndpointSuffix, d.sasTokenExpirationMinutes)
+	if genErr != nil {
+		return genErr
+	}
+
+	if _, rerr := d.cloud.GetFileShare(ctx, subsID, resourceGroupName, accountName, srcFileShareName); rerr != nil {
```

Comment:
is it necessary to get a file share call every time? if file share does not exist, copy file share would return error.

Reply:
done